### DISCUSSION TITLE
Many to Many Relationship with Pivot with multiple types

### DISCUSSION THREAD

**Discussion by lucasantoniooficial** [rebing/graphql-laravel#1038] [General]
I have a problem in my code, that I have a pivot that needs to be multiple types, I don't know if this will help me, but it was the solution I thought.

I have two pivot tables that relate to another table, but the two pivot tables use a single table.

I have a Type of service table and it relates to two pivots and the two pivots relate to a table called delegate executes.

I want it to be possible for me to access the pivot of the service type and access the delegateExecutes, but it doesn't accept multiple types.

Is there any solution for this?

I tried using Union Type.

**Comment by mfn** [comment]
I think to what you described, technically the UnionType is a solution for it. I used it quite often in my project for the general issue that "something can return multiple things".

I don't remember having this particular use-case with pivots though. Also: I don't use `SelectFields`, I use custom data loaders so the "how" the data is fetched is custom.

I'm not sure if we've a test covering this scenario. Maybe you can provide a simple self-contained example?

**Reply by lucasantoniooficial** [reply]
So I did it another way, I don't even remember the problem very well.

But I had another case here that doesn't work. I will send you model.

my request grahpQL:
`query {
    typeService(id: "996ba2a8-7801-4830-9f26-1a8565ca8c70"){
        id,
        stepProductions {
            id,
            pivot {
                id
            }
        }
    }
}`

Error: 
`"Call to undefined method App\\Models\\StepProduction::pivot()",`

Image StepProductionType:
![image](https://github.com/rebing/graphql-laravel/assets/58445514/f2011ced-8943-4215-82bb-1f5eaa88edb9)

This is very strange, I have one working exactly the same and it gives this error, isn't it strange?

**Reply by mfn** [reply]
Maybe it's a bug in the library, maybe in your code… I can't really say.

When things get more complex (and need a database, like most of the time) there are usually two ways to get help:
1. publish a public repo with instruction how to set it up and trigger the issue (e.g. using SQLite)
2. write a test for this lbrary and create the necessary models etc.

No. 2) is usually preferred, less to set up (infrastructure is already there), easier to test debug

ps: `is_relation` is (very likely) never necessary, it's the default. The purpose is to be able to turn it off (=`false`)

**Reply by lucasantoniooficial** [reply]
With is_relation false the error changes, it gives an error in the database, complicated...

**Reply by lucasantoniooficial** [reply]
How would you customize $getSelectFields?

Because since I need it to work, it's not working.
